### PR TITLE
spike: revive, 1.1.0+git20240708

### PR DESCRIPTION
**Pull request by rickliu2000**
Topic Description
-----------------

- tree-wide: extra-riscv => app-emulation
- spike: revive, 1.1.0+git20240708
    Revert "spike: drop, orphaned"
    This reverts commit 67754b8f175cb95e448b016b083ad5681520dd72.

Package(s) Affected
-------------------

- spike: 1.1.0+git20240708

Security Update?
----------------

No

Build Order
-----------

```
#buildit spike
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

**Experimental Architectures**

- [x] MIPS R6 64-bit (Little Endian) `mips64r6el`

### CODE REVIEW

File: app-emulation/spike/autobuild/prepare [hunk 1]
```diff
@@ -0,0 +1,3 @@
+CFLAGS+=" -fPIC"
+CXXFLAGS+=" -fPIC"
```
Comment (MingcongBai) [inline app-emulation/spike/autobuild/prepare]: Why? Is this still needed?

File: app-emulation/spike/autobuild/defines [hunk 2]
```diff
@@ -0,0 +1,7 @@
+PKGNAME=spike
+PKGDES="RISC-V ISA simulator"
+PKGSEC=devel
+PKGDEP="dtc"
+PKGSUG="xterm"
+
+AB_FLAGS_SPECS=0
```
Comment (MingcongBai) [inline app-emulation/spike/autobuild/defines]: Why? Is this still needed?

File: app-emulation/spike/autobuild/defines [hunk 3]
```diff
@@ -0,0 +1,7 @@
+PKGNAME=spike
+PKGDES="RISC-V ISA simulator"
+PKGSEC=devel
+PKGDEP="dtc"
+PKGSUG="xterm"
```
Comment (MingcongBai) [inline app-emulation/spike/autobuild/defines]: Where is this used?

**Review by MingcongBai — DISMISSED** [review]

File: app-emulation/spike/autobuild/prepare (hunk 1 above)
Comment (rickliu2000) [reply app-emulation/spike/autobuild/prepare]: no

Comment (rickliu2000) [reply app-emulation/spike/autobuild/prepare]: We no longer need this

**Review by jiegec — APPROVED** [review]

**Pull request merged**